packages/next-auth/src/core/types.ts

```diff
@@ -313,30 +316,87 @@ export interface CallbacksOptions<P = Profile, A = Account> {
    * [`getSession`](https://next-auth.js.org/getting-started/client#getsession) |
    *
    */
-  session: (params: {
-    session: Session
-    user: User | AdapterUser
-    token: JWT
-  }) => Awaitable<Session>
+  session: (
+    params:
+      | {
+          session: DefaultSession
```

Comment:
Was changing this to `DefaultSession` necessary? I was under the impression that the `Session` type exists so we could extend it in our own projects to add more data to it.  As documented here: https://next-auth.js.org/getting-started/typescript#extend-default-interface-properties

Like I currently have something like this in my project:

```
declare module "next-auth" {
  interface Session extends DefaultSession {
    user: {
      id: string;
      role: UserRole;
    } & DefaultSession["user"];
  }
}
```

And with the 4.21.0 this makes typescript angry about adding more data to the session.  Can we just switch this back to the `Session` type?  Or is there something I'm not understanding here.

Reply:
See my comment here: https://github.com/nextauthjs/next-auth/issues/7132#issuecomment-1493924161 open for constructive thoughts on the issue!